### ISSUE TITLE
Record type issue

### ISSUE THREAD

**Issue by AdrienLemaire** [neo4j/neo4j-javascript-driver#632]
Getting this error when using Record in my unit tests

> [tsserver 2351] [E] This expression is not constructable.
  Type 'Record' has no construct signatures.

**Driver version**: JS driver 4.1.2

### Steps to reproduce
driver.ts
```ts
import neo4j, { Driver } from "neo4j-driver";
import environment from "./environment";

const driver: Driver = neo4j.driver(
  environment.neo4j.uri,
  neo4j.auth.basic(environment.neo4j.user, environment.neo4j.password)
);
driver.verifyConnectivity().then((param) => console.dir(param));

export default driver;
```

simplified User.ts
```ts
const generateUserModel = (user: JwtUser): TUserModel => ({
  canUpdateRoleFor: async ({ id, role }) => {
    const session = driver.session();
    try {
      const result = await session.run(
        "MATCH (u:User {id: $id}) RETURN u.role",
        { id }
      );
      const currentRole: Role = result.records[0].get(0);
      return currentRole
    } finally {
      await session.close();
    }
```

User.spec.ts
```ts
import { types } from "neo4j-driver";
const { Record } = types; // Strange, but I can't import Record directly from neo4j-driver

jest.mock("../driver", () => {
  // Mock driver.session() used in canUpdateRoleFor
  return {
    __esModule: true,
    default: {
      session: jest.fn(() => ({
        run: jest.fn((_cypher, { id }) => {
          return {
            records: [new Record(["role"], [SortedRoles[id]])], // <= LINE RAISING THE TYPE ERROR
          };
        }),
        close: jest.fn(),
      })),
    },
  };
});
```

### Expected behavior

No type error

### Actual behavior

![](https://i.imgur.com/K30PQTJ.png)

I verified that the implementation is correct and my tests pass as expected.

Also, I cannot find the reason for this error in the code. When jumping to the type definition, I can see a constructor definition.

neo4j-driver/types/record.d.ts
```ts
declare class Record {
  keys: string[]
  length: number

  constructor(
    keys: string[],
    fields: any[],
    fieldLookup?: { [index: string]: string }
  )
```

**Comment by bboure** [comment]
I am facing the exact same issue.
I think the issue is [here](https://github.com/neo4j/neo4j-javascript-driver/blob/4.3/types/index.d.ts#L101)

it should be 
```ts
  Record: typeof Record
```

**Comment by bigmontz** [comment]
Hi @bboure  and @AdrienLemaire,

It's the mismatch between the `index.js` and the `index.d.ts`. Alternatively, you could instantiate the record like code bellow:

```
import { types, Record } from "neo4j-driver"
const record: Record = new types.Record(['name', 'age'], ['Bob', 22])
const value = record.get('name')
```

**Comment by bboure** [comment]
@bigmontz Thanks for getting back.
This is what I did. The error is still there.

**Comment by bigmontz** [comment]
@bboure some environment configurations i am using to run the script. 

tsconfig.json
```
{
    "compilerOptions": {
        "noImplicitAny": true,
        "noImplicitReturns": true,
        "strictNullChecks": true,
        "moduleResolution": "node",
        "types": []
    }
}
```
package.json
```
 "scripts": {
    "dev": "ts-node-dev --transpile-only --ignore-watch node_modules src/server.ts"
  },
  "devDependencies": {
    "ts-node-dev": "^1.0.0-pre.63",
    "typescript": "^4.0.3"
  },
```

**Comment by bigmontz** [comment]
It will be solved in the 4.3

**Comment by AdrienLemaire** [comment]
@bigmontz looking forward to 4.3, as it'll finally allow me to write unittests with a mocked driver.
